components/fxa-client/examples/migration.rs

```diff
@@ -15,8 +14,22 @@ fn main() {
     let k_sync: String = prompt_string("k_sync").unwrap();
     println!("Enter kXCS (hex-string):");
     let k_xcs: String = prompt_string("k_xcs").unwrap();
-    fxa.migrate_from_session_token(&session_token, &k_sync, &k_xcs, true)
-        .unwrap();
-    println!("WOW! You've been migrated.");
+    let migration_result =
+        match fxa.migrate_from_session_token(&session_token, &k_sync, &k_xcs, true) {
+            Ok(migration_result) => migration_result,
+            Err(err) => {
+                println!("Error: {}", err);
+                // test offline behaviour
+                loop {
+                    thread::sleep(time::Duration::from_millis(5000));
+                    let retry = fxa.try_migration();
+                    match retry {
+                        Ok(result) => break result,
+                        Err(_) => println!("Retrying... Are you connected to the internet?"),
```

Comment:
Continuing discussion from [the previous PR](https://github.com/mozilla/application-services/pull/2481#pullrequestreview-344317428), each migration attempt could fail for either both fatal or non-fatal reasons. It would be useful for this example to show how to handle the two different cases. For example right now, if I enter an invalid sessionToken into this example script, then IIUC it will loop forever asking me whether I'm connected to the internet. We don't want to get anyone's migrated Fennec stuck in such a loop in practice.

Reply:
(I'll also note that "not connected to the internet" is not necessarily the only transient error; 500-level server errors should probably also be treated as transient and retried).

Reply:
I know it's only for testing purposes, but I'd like to see this example use the same logic as we expect consumers to use in practice, which IIUC is basically:

```
match migrate_from_session_token(...) {
  Ok(result) => result,
  Err(...) => {
    while is_in_migration_state() {
       sleep(...)
       match retry_migrate_from_session_token() {
         Ok(result) => result,
         Err(...) => continue,
        }
    }
  }
}
```